core/unix/coredump.c

```diff
@@ -51,6 +51,7 @@
 #    error Unsupported architecture
 #endif
 
+#define MAX_OS_PAGE_SIZE 4096
```

Comment:
This will fail on some of our Arm machines: best to set this to at least 16KB; or, use a dynamically-allocated buffer.

Reply:
I've increased the max to 16384.

Reply:
We may have seen machines with 64K?  The commits updating the options that depend on page size may say. 
 Probably this has to be dynamic after all; could either loop over a 4K buffer (that sounds like a good compromise) or dynamically allocate the whole thing.

Reply:
Let's not have a hardcoded assumption here mean this fails to run later on some arm machine.

Reply:
I've removed the hardcoded number and changed the code to loop through a 4k buffer.